scripts/extractions/extract_common.py

```diff
@@ -0,0 +1,191 @@
+"""Common functions used by extraction scripts."""
+
+import json
+import logging
+import os
+from datetime import datetime
+from importlib.metadata import version
+from pathlib import Path
+from tempfile import NamedTemporaryFile
+from typing import List
+
+import geojson
+import geopandas as gpd
+import pandas as pd
+import pystac
+import requests
+from openeo_gfmap.utils.netcdf import update_nc_attributes
+from shapely import Point
+
+# Logger used for the pipeline
+pipeline_log = logging.getLogger("pipeline_sar")
+
+pipeline_log.setLevel(level=logging.INFO)
+
+stream_handler = logging.StreamHandler()
+pipeline_log.addHandler(stream_handler)
+
+formatter = logging.Formatter("%(asctime)s|%(name)s|%(levelname)s:  %(message)s")
+stream_handler.setFormatter(formatter)
+
+
+# Exclude the other loggers from other libraries
+class ManagerLoggerFilter(logging.Filter):
+    """Filter to only accept the OpenEO-GFMAP manager logs."""
+
+    def filter(self, record):
+        return record.name in [pipeline_log.name]
+
+
+stream_handler.addFilter(ManagerLoggerFilter())
+
+
+def post_job_action(
+    job_items: List[pystac.Item], row: pd.Series, parameters: dict = {}
+) -> list:
+    """From the job items, extract the metadata and save it in a netcdf file."""
+    base_gpd = gpd.GeoDataFrame.from_features(json.loads(row.geometry)).set_crs(
+        epsg=4326
+    )
+    assert len(base_gpd[base_gpd.extract == 2]) == len(
+        job_items
+    ), "The number of result paths should be the same as the number of geometries"
+
+    extracted_gpd = base_gpd[base_gpd.extract == 2].reset_index(drop=True)
+    # In this case we want to burn the metadata in a new file in the same folder as the S2 product
+    for idx, item in enumerate(job_items):
+        if "sample_id" in extracted_gpd.columns:
+            sample_id = extracted_gpd.iloc[idx].sample_id
+        else:
+            sample_id = extracted_gpd.iloc[idx].sampleID
+
+        ref_id = extracted_gpd.iloc[idx].ref_id
+        valid_time = extracted_gpd.iloc[idx].valid_time
+        h3_l3_cell = extracted_gpd.iloc[idx].h3_l3_cell
+        s2_tile = row.s2_tile
+
+        item_asset_path = Path(list(item.assets.values())[0].href)
+
+        # Add some metadata to the result_df netcdf file
+        new_attributes = {
```

Comment:
@GriffinBabe not sure if I fully understand this, but the `extract_common.py` seems to hold some common functionality like post job action, but the default parameters below are all related to S1. What's the reason for this? If some script does not provide a required parameter, the S1 information will be used? Won't that lead to confusion?

Reply:
Yes indeed, makes sense I'm changing this